core/arch/arm/kernel/thread_spmc.c

```diff
@@ -2641,13 +2647,16 @@ static void *spmc_retrieve_req(struct ffa_mem_transaction_x *trans)
 		trans_descr->flags = trans->flags;
 		trans_descr->mem_access_count = 1;
 		trans_descr->mem_access_offs = sizeof(*trans_descr);
-		trans_descr->mem_access_size = sizeof(struct ffa_mem_access);
-		acc_descr_array = (void *)((vaddr_t)my_rxtx.tx +
-					   sizeof(*trans_descr));
+		trans_descr->mem_access_size = mem_acc_size;
+		mem_acc = (void *)((vaddr_t)my_rxtx.tx + sizeof(*trans_descr));
 	}
-	acc_descr_array->region_offs = 0;
-	acc_descr_array->reserved = 0;
-	perm_descr = &acc_descr_array->access_perm;
+	/*
+	 * We're using the FF-A 1.1 endpoint memory access descriptor even
+	 * where we should have used the FF-A 1.2 version since we only
+	 * access the first two common fields.
+	 */
```

Comment:
Strictly speaking, `mem_acc` is a `ffa_mem_access_common` not a 1.1 descriptor, so the comment is kind of obvious.